src/FixedDataTable.js

```diff
@@ -508,43 +508,77 @@ class FixedDataTable extends React.Component {
     this._didScrollStop = debounceCore(this._didScrollStopSync, 200, this);
     this._onKeyDown = this._onKeyDown.bind(this);
 
-    this._wheelHandler = new ReactWheelHandler(
-      this._onScroll,
-      this._shouldHandleWheelX,
-      this._shouldHandleWheelY,
-      this.props.isRTL,
-      this.props.stopScrollDefaultHandling,
-      this.props.stopScrollPropagation
-    );
-
-    this._touchHandler = new ReactTouchHandler(
-      this._onScroll,
-      this._shouldHandleTouchX,
-      this._shouldHandleTouchY,
-      this.props.stopScrollDefaultHandling,
-      this.props.stopScrollPropagation
-    );
+    this._setupHandlers();
   }
 
   componentWillUnmount() {
+    this._cleanupHandlers();
+
+    // Cancel any pending debounced scroll handling and handle immediately.
+    this._didScrollStop.reset();
+    this._didScrollStopSync();
+  }
+
+  _setupHandlers() {
+    if (!this._wheelHandler) {
```

Comment:
I added guard checks to ensure we won't create handlers more than once.

This lets us call _setupHandlers multiple times.